libfreerdp/utils/smartcard_pack.c

```diff
@@ -342,33 +342,24 @@ static char* smartcard_convert_string_list(const void* in, size_t bytes, BOOL un
 
 	if (unicode)
 	{
-		length = (bytes / sizeof(WCHAR)) - 1;
-		WINPR_ASSERT(length < INT_MAX);
-
-		mszA = (char*)calloc(length + 1, sizeof(char));
+		mszA = ConvertMszWCharNToUtf8Alloc(string.wz, bytes / sizeof(WCHAR), &length);
 		if (!mszA)
 			return NULL;
-		if (ConvertFromUnicode(CP_UTF8, 0, string.wz, (int)length, &mszA, (int)length + 1, NULL,
-		                       NULL) != (int)length)
-		{
-			free(mszA);
-			return NULL;
-		}
 	}
 	else
 	{
-		length = bytes;
-		mszA = (char*)calloc(length, sizeof(char));
+		mszA = (char*)calloc(bytes, sizeof(char));
 		if (!mszA)
 			return NULL;
-		CopyMemory(mszA, string.sz, length - 1);
-		mszA[length - 1] = '\0';
+		CopyMemory(mszA, string.sz, bytes - 1);
+		mszA[bytes - 1] = '\0';
+		length = bytes - 1;
 	}
 
-	for (index = 0; index < length - 1; index++)
+	for (index = 1; index < length; index++)
```

Comment:
Why not start with `index = 0` and use `mszA[index]`?

Reply:
good catch